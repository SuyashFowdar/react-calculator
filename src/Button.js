import React from 'react';
import PropTypes from 'prop-types';

const Button = props => {
  const { numbers, symbols } = props;
  const numberList = numbers.map(number => <button type="button" key={number.toString()}>{number}</button>);
  const symbolList = symbols.map(symbol => <button type="button" key={symbol.toString()}>{symbol}</button>);

  return (
    <div>
      <div>{numberList}</div>
      <div>{symbolList}</div>
    </div>
  );
};

Button.propTypes = {
  numbers: PropTypes.number.isRequired,
  symbols: PropTypes.string.isRequired,
};

export default Button;
