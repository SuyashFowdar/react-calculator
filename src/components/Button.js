import React from 'react';
import PropTypes from 'prop-types';

const Button = props => {
  const {
    name,
    handleClick,
    color,
    wide,
  } = props;

  return (
    <button type="button" onClick={() => handleClick(name)} style={{ backgroundColor: color, width: wide ? '50%' : '25%' }}>
      {name}
    </button>
  );
};

Button.defaultProps = {
  color: 'orange',
  wide: false,
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  color: PropTypes.string,
  wide: PropTypes.bool,
};

export default Button;
