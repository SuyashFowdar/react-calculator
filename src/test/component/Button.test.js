import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

it('renders correctly', () => {
  const isWide = true;
  const tree = renderer
    .create(<Button name="0" key="0" handleClick={() => {}} wide={isWide} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
