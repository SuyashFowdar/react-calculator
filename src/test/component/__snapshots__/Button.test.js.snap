// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "orange",
      "width": "50%",
    }
  }
  type="button"
>
  0
</button>
`;
